addons/mail/static/src/components/composer_text_input/composer_text_input.js

```diff
@@ -65,6 +65,22 @@ class ComposerTextInput extends Component {
         return this.env.models['mail.composer'].get(this.props.composerLocalId);
     }
 
+    /**
+     * @returns {string}
+     */
+    get textareaPlaceholder() {
+        if (!this.composer) {
+            return "";
+        }
+        if (this.composer.thread && this.composer.thread.model === 'mail.channel') {
+            return this.env._t("Write something...");
+        }
+        if (this.composer.isLog) {
+            return this.env._t("Log an internal note...");
+        }
+        return this.env._t("Send a message to followers...");
```

Comment:
I would expect fallback "Write something...", because the 2 other strings are specific to chatter.

Reply:
I was considering this but at the end I kind of disagree.
The basic thread is a non-channel thread, so that makes sense.

The channel thread is a specific case.

Reply:
What if you don't have a thread? This doesn't make sense to display "Send a message to followers..."
While "Write something..." is kinda the placeholder that fits all input...

Reply:
I don't think we support that right now (and I don't know if we ever will), but if so, a no-thread composer would also be a special case to me and there would be a lot more issues than just this placeholder to resolve.

Reply:
PS: I don't actually care what we do here, both are acceptable, it's just a matter of point of view.

Reply:
2 tests in `component/composer_tests.js` were creating stand-alone composers and made assertions with the 2 chatter-specific strings. Those tests should never have passed if the "Write something..." string was the fallback, and it makes totally sense to it be the default.

Pre-emptive `return` are good to reduce nested scopes, which eases readability. But doing so, there are good practices: use conditionals for unusual/special outcomes, and have the basic/simplest/less contextual outcome as the final `return`. The implementation detail of this function would be better if we follow this by having "Write something..." as the fallback.

Reply:
As I said, it's true too, but it depends on point of view. You consider "Write something..." as the default placeholder, while I consider "thread is not a channel" as the default non contextual condition.

Anyway as I said, I'm fine with the change too, I'm just highlighting that it is not that obvious.